.github/workflows/s3-tests.yml

```diff
@@ -0,0 +1,318 @@
+        with:
+          repository: nspcc-dev/neofs-s3-dev-env
+          ref: master
+          path: neofs-s3-dev-env
+
+      - name: Download latest stable neofs-cli for uploading reports to NeoFS
+        uses: dsaltares/fetch-gh-release-asset@1.1.1
+        with:
+          repo: 'nspcc-dev/neofs-node'
+          version: 'tags/v0.37.0'
+          file: 'neofs-cli-amd64'
+          target: 'neofs-node-stable/neofs-cli'
+
+      - name: Checkout neofs-testcases repository
+        uses: actions/checkout@v3
+        with:
+          repository: nspcc-dev/neofs-testcases
+          ref: 'master'
+          path: neofs-testcases
+
+################################################################
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          cache: true
+          go-version: '1.20'
+      - run: go version
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - run: python --version
+
+################################################################
+      - name: Get TAG for docker images
+        run: |
+          echo "CURRENT_TAG=$( make version | sed 's/^v//' )" >> $GITHUB_ENV
+        working-directory: neofs-s3-gw
+
+################################################################
+
+      - name: Build neofs-s3-gw docker image
+        run: |
+          cp .github/test-env .env #looks ugly, but I want to try
+          make
+        working-directory: neofs-s3-gw
+
+      - name: Build neofs-s3-gw binaries
+        run: |
+          make image
+        working-directory: neofs-s3-gw
+
+
+#################################################################
+
+      - name: Add NeoFS S3-gw TAGs to s3-gw-test env config
+        run: |
+          sed -i -e 's/S3_GW_VERSION=.*$/S3_GW_VERSION=${{ env.CURRENT_TAG }}/' .github/test-env
+
+          echo "####### DEBUGGING INFO: ##############"
+          realpath .github/test-env
+          cat .github/test-env
+        working-directory: neofs-s3-gw
+
+      - name: Copy test-env file to .env for neofs-dev-env
+        run: |
+          cp .github/test-env ${GITHUB_WORKSPACE}/neofs-dev-env/.env
+        working-directory: neofs-s3-gw
+
+################################################################
+
+      - name: Prepare hosts
+        timeout-minutes: 5
+        run: |
+          make get
+          sudo ./bin/update_hosts.sh
+          sudo chmod a+w vendor/hosts
+        working-directory: neofs-dev-env
+
+      - name: Prepare Dev-Env to run tests
+        id: prepare_test_env
+        timeout-minutes: 30
+        run: |
+          make prepare-test-env
+          echo "$(pwd)/vendor" >> $GITHUB_PATH
+        working-directory: neofs-dev-env
+
+      - name: Set zero fee
+        run: |
+          make update.container_fee val=0 && make update.container_alias_fee val=0
+        working-directory: neofs-dev-env
+
+      - name: Copy neofs-s3-gw binaries to vendor directory
```

Comment:
Shouldn't this be done before `make prepare-test-env` ?

Reply:
As far as I see it doesn't, as I build s3-gw container from the latest code and not using the release one from dev-env.